tests/test_mixed_simulator_device.py

```diff
@@ -0,0 +1,910 @@
+# Copyright 2018 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for the MixedStateMixedStateSimulatorDevice
+"""
+import pytest
+import math
+
+import pennylane as qml
+import numpy as np
+from pennylane_cirq import MixedStateSimulatorDevice
+import cirq
+
+
+class TestDeviceIntegration:
+    """Tests that the MixedStateSimulatorDevice integrates well with PennyLane"""
+
+    def test_device_loading(self):
+        """Tests that the cirq.simulator device is properly loaded"""
+
+        dev = qml.device("cirq.mixedsimulator", wires=2)
+
+        assert dev.num_wires == 2
+        assert dev.shots == 1000
+        assert dev.short_name == "cirq.mixedsimulator"
+
+        assert isinstance(dev, MixedStateSimulatorDevice)
+
+
+@pytest.fixture(scope="function")
+def simulator_device_1_wire(shots, analytic):
+    """Return a single wire instance of the MixedStateSimulatorDevice class."""
+    yield MixedStateSimulatorDevice(1, shots=shots, analytic=analytic)
+
+
+@pytest.fixture(scope="function")
+def simulator_device_2_wires(shots, analytic):
+    """Return a two wire instance of the MixedStateSimulatorDevice class."""
+    yield MixedStateSimulatorDevice(2, shots=shots, analytic=analytic)
+
+
+@pytest.fixture(scope="function")
+def simulator_device_3_wires(shots, analytic):
+    """Return a three wire instance of the MixedStateSimulatorDevice class."""
+    yield MixedStateSimulatorDevice(3, shots=shots, analytic=analytic)
```

Comment:
Nitpick: I know that this is a copy and paste, but I think these fixtures are redundant/overkill, we should simply instantiate the device in the test.

Ideally, tests should be as self contained as possible where it makes sense. Test data is fine to be parametrized, whereas logical units shouldn't be parametrized. This also requires a lot of scrolling around the test file to gather up the test components.

A test that can be copied and pasted into a Python console for debugging, and works as-is, is 👨‍🍳 in my books.

Reply:
Maybe next time :cold_sweat: